airflow/exceptions.py

```diff
@@ -29,6 +29,12 @@
     from airflow.models import DagRun
 
 
+class AirflowKillSignal(Exception):
+    """Raise when there's kill signal"""
```

Comment:
```suggestion
class AirflowTermSignal(Exception):
    """Raise when we receive a TERM signal"""
```

We should consider using something other than "kill" here, as a true SIGKILL won't hit this.